File: pkg_roscpp/src/publisher_msg.cpp
#include "ros/ros.h"
#include "pkg_roscpp/simple_msg.h"

int main(int argc, char **argv){
    ros::init(argc, argv, "publisher");
    ros::NodeHandle node_obj;
    ros::Publisher publisher_obj = node_obj.advertise<pkg_roscpp::simple_msg>("/topic_publisher_msg", 10);
    ros::Rate loop_rate(10);
    
    pkg_roscpp::simple_msg msg;
    uint8_t bool_val = true;
    int32_t num_val = 0;  //int32 in ros msg corresponds to int32_t of c++
    uint32_t unsigned_num_val = 10;
    float float_val = 0.0f;
    double double_val = 0.0;
    std::string text = "Hello World";   //string corresponds to std::string (c++ style string)

    while (ros::ok()){
        msg.boolean = bool_val;
        msg.num = num_val;
        msg.unsigned_num = unsigned_num_val;
        msg.float_num = float_val;
        msg.double_num = double_val;
        msg.str = text;

        ROS_INFO("%d",msg.num);
        ROS_INFO("%s", (msg.str).c_str());  //convert c++ string to c string for printing
        publisher_obj.publish(msg);
        
        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: pkg_rospy/scripts/publisher_msg.py
#!/usr/bin/env python

import rospy
from pkg_rospy.msg import simple_msg

def main():
    rospy.init_node('msg_publisher_py', anonymous=False)
    pub = rospy.Publisher('/topic_publisher_msg', simple_msg, queue_size=10)
    rate = rospy.Rate(10)
    msg = simple_msg()
    
    while not rospy.is_shutdown():
        msg.boolean = False
        msg.num = 100
        msg.unsigned_num = 50000
        msg.float_num = 10.0
        msg.double_num = 50.0
        msg.str = "Hello"
        pub.publish(msg)
        rospy.loginfo("Custom message sent")
        rate.sleep()

if __name__ == '__main__':
    try:
        main()
    except rospy.ROSInterruptException:
        pass